### ISSUE TITLE
max-duration does not stop transfers if retries > 1 are specified

### ISSUE THREAD

**Issue by DiscreteObject** [rclone/rclone#6002]
#### The associated forum post URL from `https://forum.rclone.org`
https://forum.rclone.org/t/max-duration-and-retries-not-working-correctly/27738/14

#### What is the problem you are having with rclone?
`--max-duration` does not stop jobs as expected but instead triggers retries up to the `--retries` specified. This seems unexpected from the docs description of retries:
```
--retries int
Retry the entire sync if it fails this many times it fails (default 3).
```
as I would not consider hitting the specified `max-duration` a failure. The docs for `max-duration` also don't indicate this:
```
--max-duration=TIME
Rclone will stop scheduling new transfers when it has run for the duration specified.
Defaults to off.
When the limit is reached any existing transfers will complete.
```

This seems related to the forum post above (not posted by me), and to these previously filed issues:  https://github.com/rclone/rclone/issues/4504 and https://github.com/rclone/rclone/issues/4002. However the forum post indicates it should be fixed with the latest 1.58.0 beta, and even with that I still see the issue.

I've included a debug log demonstrating this, but to summarize what I think are the relevant parts: a copy job to an encrypted Backblaze B2 remote with `"--max-duration=5m"` and `"--retries" "3"`.  The job begins by setting a deadline 5 minutes out:
```
2022/02/19 11:01:17 INFO  : Encrypted drive 'b2cryptsse:max_duration_test/Photos/archives': Transfer session deadline: 2022/02/19 11:06:17
```
At the 5-minute mark, the logs indicate it has hit the deadline but then begins a repeat of the same job with a new deadline another 5 minutes later, also doing file comparisons of the files that had already been uploaded:
```
2022/02/19 11:06:17 ERROR : photos_2021.zip.004: Failed to copy: Post "https://pod-040-2006-14.backblaze.com/b2api/v1/b2_upload_file/f433ac23c06708ee77f1021b/c004_v0402006_t0014": context deadline exceeded
2022/02/19 11:06:17 ERROR : photos_2021.zip.002: Failed to copy: Post "https://pod-040-2005-03.backblaze.com/b2api/v1/b2_upload_file/f433ac23c06708ee77f1021b/c004_v0402005_t0004": context deadline exceeded
2022/02/19 11:06:17 ERROR : photos_2021.zip.001: Failed to copy: Post "https://pod-040-2002-15.backblaze.com/b2api/v1/b2_upload_file/f433ac23c06708ee77f1021b/c004_v0402002_t0057": context deadline exceeded
2022/02/19 11:06:17 ERROR : photos_2021.zip.003: Failed to copy: Post "https://pod-040-2001-07.backblaze.com/b2api/v1/b2_upload_file/f433ac23c06708ee77f1021b/c004_v0402001_t0015": context deadline exceeded
2022/02/19 11:06:17 ERROR : photos_2021.vol063+64.par2: Failed to copy: Post "https://pod-040-2000-16.backblaze.com/b2api/v1/b2_upload_file/f433ac23c06708ee77f1021b/c004_v0402000_t0011": context deadline exceeded
2022/02/19 11:06:17 ERROR : Attempt 1/3 failed with 5 errors and: Post "https://pod-040-2002-15.backblaze.com/b2api/v1/b2_upload_file/f433ac23c06708ee77f1021b/c004_v0402002_t0057": context deadline exceeded
2022/02/19 11:06:17 INFO  : Encrypted drive 'b2cryptsse:max_duration_test/Photos/archives': Transfer session deadline: 2022/02/19 11:11:17
2022/02/19 11:06:18 DEBUG : photos_2021.par2: Size and modification time the same (differ by -66.5µs, within tolerance 1ms)
```
The same thing 5 minutes later:
```
2022/02/19 11:11:17 ERROR : Attempt 2/3 failed with 5 errors and: Post "https://pod-040-2006-14.backblaze.com/b2api/v1/b2_upload_file/f433ac23c06708ee77f1021b/c004_v0402006_t0014": context deadline exceeded
2022/02/19 11:11:17 INFO  : Encrypted drive 'b2cryptsse:max_duration_test/Photos/archives': Transfer session deadline: 2022/02/19 11:16:17
```
And then having hit 3 retries, it finally fails 5 minutes after that
```
2022/02/19 11:16:17 ERROR : Attempt 3/3 failed with 5 errors and: Post "https://pod-040-2005-03.backblaze.com/b2api/v1/b2_upload_file/f433ac23c06708ee77f1021b/c004_v0402005_t0004": context deadline exceeded
2022/02/19 11:16:17 NOTICE: 
Transferred:   	    3.842 GiB / 12.761 GiB, 30%, 4.386 MiB/s, ETA 34m42s
Errors:                 5 (retrying may help)
Checks:                16 / 16, 100%
Transferred:            8 / 18, 44%
Elapsed time:      15m0.4s
```

It seems like I could avoid this by setting `retries=1` but I'd like to be able to specify retries in the case of something like network disruption, but also set a max duration if I want the job to run a set amount of time.

#### What is your rclone version (output from `rclone version`)
```
rclone v1.58.0-beta.6000.dc7e3ea1e
- os/version: Microsoft Windows 7 Professional Service Pack 1 (64 bit)
- os/kernel: 6.1.7601.24544 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.17.6
- go/linking: dynamic
- go/tags: cmount
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Windows 7 x64 SP1

#### Which cloud storage system are you using? (e.g. Google Drive)
Backblaze B2

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
```
>rclone.exe copy --transfers 5 --checkers 8 --contimeout 60s --timeout 300s --retries 3 --low-level-retries 10 --b2-chunk-size=100M  --b2-upload-cutoff=1025M --log-file=rclone_photos_archives_158_beta.log --log-level=DEBUG --progress --stats 1s --stats-file-name-length 0 --stats-log-level=NOTICE --max-duration=5m --fast-list "D:\Photos\archives" "b2cryptsse:max_duration_test/Photos/archives"
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
https://pastebin.com/SgnEbQjj

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by DiscreteObject** [comment]
Thanks for looking into this!

Are any of those builds for Windows x64? It doesn't look like it from the filenames and that's what I use primarily.

**Comment by ncw** [comment]
Sorry, the tests failed so the build failed! I've fixed up the tests now - can you have a go with this

[v1.58.0-beta.6009.eff64e528.fix-6002-max-duration](https://beta.rclone.org/branch/fix-6002-max-duration/v1.58.0-beta.6009.eff64e528.fix-6002-max-duration/) on branch [fix-6002-max-duration](https://github.com/rclone/rclone/tree/fix-6002-max-duration) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.58
